src/api/valence-api.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const { LEVersion, LPVersion } = require('../constants');
+
+module.exports = class ValenceApi {
+	constructor(
+		valence,
+		fetch = require('node-fetch')
+	) {
+		this._valence = valence;
+		this._fetch = fetch;
+	}
+
+	async getContent(instanceUrl, orgUnit, module = null) {
+		const url = module
+			? new URL(`/d2l/api/le/${LEVersion}/${orgUnit.Identifier}/content/modules/${module.Id}/structure/`, instanceUrl)
+			: new URL(`/d2l/api/le/${LEVersion}/${orgUnit.Identifier}/content/root/`, instanceUrl);
+		const signedUrl = this._valence.createAuthenticatedUrl(url, 'GET');
+
+		const response = await this._fetch(signedUrl);
+
+		this._assertResponse(response);
+
+		return response.json();
+	}
+
+	async createModule(instanceUrl, orgUnit, module, parentModule = null) {
```

Comment:
I think it's worthwhile to make the `module` parameter only contain the properties that we care about to ensure there's an abstraction between our code and Valence.  This method can map what we care about into the structure that Valence expects.
IsLocked and ModuleStartDate are examples of properties valence expects that we don't care about.